### ISSUE TITLE
[BUG]: Sorting is not working

### ISSUE THREAD

**Issue by schngupta** [microsoft/powercat-creator-kit#454]
**Describe the bug**
When I open my Canvas app or Creator Kit Reference App they showed me warning message is "Disable component behavior property return type check" is on because one or more component behavior properties, event or action, have mismatched return types." due to that reason I anticipate sorting is stopped working for all columns except number columns. 

**To Reproduce**
1. Edit latest creator kit reference app in PowerApps Studio
2. go to DetailsList control 
3. try to sort any of the column
4. nothing works

**Expected behavior**
Sorting should work if that is set as sortable column
**Screenshots**
![image](https://github.com/microsoft/powercat-creator-kit/assets/8862555/cbb18091-681a-4c26-b15f-94cace38b5ff)

**Comment by Jenefer-Monroe** [comment]
I repro this also in the CoE Starter Kit.

**Comment by Jenefer-Monroe** [comment]
same bug repros in modern grid control

**Comment by Lou-i3** [comment]
Same issue with the latest release of the Creator Kit. 
I tried enabling the feature "enhanced component properties": the warning message is gone but sorting still doesn't work.

**Comment by lbuijtelaar** [comment]
For me in a canvas app & SQL source the numeric columns can be sorted, but not the string columns.

**Comment by seebeelim** [comment]
I have the same issue when the powerapp release version changed from v 3.24032.23 to v3.23032.26 on 22 Mar 2024 afternoon. I can't get more component from my app anymore. Is there any workaround?

**Comment by Jenefer-Monroe** [comment]
Thank you everyone. Sorry for the radio silence from CAT here. 
We do have an internal escalation posted for this. Please keep new information / findings coming here as they will help.

**Comment by markgomez00** [comment]
I have the same issue.

Suddenly text/number column can't be sorted even if it's sortable true.

Choice and lookup columns are working fine

**Comment by Jenefer-Monroe** [comment]
issue is understood by dev, can't back out the change due to other reasons but they are actively working on a fix hopefully in the next week.

**Comment by AkshayBobade03** [comment]
It seems, sorting is still working in the old published app but not in Power Apps editor mode.

**Comment by Jenefer-Monroe** [comment]
correct. until you save and publish things are fine. But once you save and publish you are broken.

**Comment by cshiflett** [comment]
Ironically, I am experiencing the same issue with the new modern table control.

**Comment by Jenefer-Monroe** [comment]
Thank you @cshiflett, I have let the dev know that it impacts the new control as well. 
Fingers are crossed that this fix rolls early this week, I'll let you all know ASAP

**Comment by miberr** [comment]
Same issue in custom page, authoring version 3.24042.21

**Comment by Jenefer-Monroe** [comment]
This is resolved for my test cases now. YEAH! 
Note that you do need to open the app in edit mode to fix (just like you had to open in edit mode to break).

**Comment by SeeBeeDev** [comment]
Great. Thanks Jenefer. I would like to check which release version of PowerApp? I'm using 3.24042.24 and managed to get the sorting working on default columns. Whereas it gave an error when we are using "AddColumn" column where it would gave a warning message on top "The specified column "columnname" does not exist or in an invalid sort column type.

**Comment by markgomez00** [comment]
It is now working after re-publishing the page on my environment. Thanks @Jenefer-Monroe

**Comment by Jenefer-Monroe** [comment]
> Great. Thanks Jenefer. I would like to check which release version of PowerApp? I'm using 3.24042.24 and managed to get the sorting working on default columns. Whereas it gave an error when we are using "AddColumn" column where it would gave a warning message on top "The specified column "columnname" does not exist or in an invalid sort column type.

I’m afraid I’m not a creator kit owner and so can’t answer more questions, I was just impacted by this bug for the CoE Starter Kit
I would suggest you post a new issue and hopefully sobering from creator Kit can assist
